src/hb-serialize.hh

```diff
@@ -70,6 +70,40 @@ struct hb_serialize_context_t
       virtual_links.fini ();
     }
 
+    object_t () = default;
+
+    object_t (const cy_object_t &o)
+    {
+      head = o.head;
+      tail = o.tail;
+      next = nullptr;
+      real_links.alloc (o.num_real_links);
+      for (unsigned i = 0 ; i < o.num_real_links; i++)
+      {
+        link_t l;
```

Comment:
You could add a copy constructor to link_t that takes a cy_link_t then this copy code doesn't need to be duplicated here and in the next loop